scripts/build-sidecar.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -ex
+if [ "${SKIP_SIDECAR_BUILD}" == "true" ]; then
+  echo "Skipping Sidecar Build because SKIP_SIDECAR_BUILD was set to 'true'"
+else
+  SCRIPT_DIR=$( dirname -- "$( readlink -f -- "$0"; )"; )
+  SIDECAR_REPO="${SIDECAR_REPO:-https://github.com/apache/cassandra-sidecar.git}"
+  SIDECAR_BRANCH="${SIDECAR_BRANCH:-trunk}"
```

Comment:
You can specify any `commit-like` thing you want for `SIDECAR_BRANCH` - I suppose it's a bit misleadingly named as it would take a SHA or tag as well. If we want to fix these things in the CI build we could just provide SHAs in the circle config, but I'm a bit concerned about making sure we're actually doing _continuous integration_ if we are fixing SHAs rather than using `trunk:head` - because these two projects are so inter-dependent _not_ doing continuous integration against trunk will likely cause us some pain as the trunk drifts away from released software.

Reply:
After an off-line discussion, pulled in changes similar to `build-dtest-jars.sh` so we can specify a SHA to build with.